chia/consensus/blockchain.py

```diff
@@ -410,9 +410,7 @@ async def _reconsider_peak(
                 tx_additions,
                 tx_removals,
             )
-            removed_rec: List[Optional[CoinRecord]] = [
-                await self.coin_store.get_coin_record(name) for name in tx_removals
-            ]
+            removed_rec: List[CoinRecord] = await self.coin_store.get_coin_records(tx_removals)
```

Comment:
there's an issue here. The assert on line 422 need to be removed and instead an assert here needs to be:

```
assert len(removed_rec) == len(tx_removals)
```